### PR TITLE
Added selected class to data provider options.

### PR DESCRIPTION
**Pull request by ConorWebb96**
## Description
When lots of data providers/queries/views are on a page. Its very difficult to see what is currently selected in the list. This highlights whatever is selected in blue.

Addresses: 
- [Source query highlight at Provider data dropdown stopped working](https://github.com/Budibase/budibase/discussions/12132)

## Screenshots
Screenshot 2023-10-27 at 11 01 51
Screenshot 2023-10-27 at 11 02 21

## Documentation
- [x] I have reviewed the Budibase documentation to verify if this feature requires any changes. If changes or new docs are required I have written them.

## Feature branch env
[Feature Branch Link](http://fb-add-select-states-to-dropdown-data-provider-select.fb.qa.budibase.net)

### CODE REVIEW

**Review by deanhannigan — CHANGES_REQUESTED** [review]
Hey Conor, the code changes look great but would it be possible to change the design to be more in line with other selects in the builder? Typically there is a small tick to indicate the current selection.
 
Something like the `Edit Column` modal in the data section:

Screenshot 2023-10-27 at 14 48 44

**Comment by aptkingston** [comment]
The main reason I haven't implemented this before is that we don't actually have a consistent way to identify a given datasource - they all have different properties. This PR uses the label, but that would mean that if you had a table named `Foo`, and a query named `Foo` then they would both be marked as selected.

The only real way to tell is compare both the type and identification property of each datasource against each other. You can't even just do a deep compare because schema etc may have changed. So that means comparing `_id` for tables, `id` for view V2s, `providerId` for data providers etc. It's quite a lot of conditional checks.

The easier approach is probably add some sort of new `key` property to all options, and use this going forward to determine what's selected, just accepting that existing values will not be marked as selected until they are changed once more.

**Comment by ConorWebb96** [comment]
Thanks for the feedback @deanhannigan,

How does this look?
Screenshot 2023-10-30 at 10 03 49

**Review by deanhannigan — APPROVED** [review]

**Review by aptkingston — APPROVED** [review]

**Comment by aptkingston** [comment]
There's a lot of duplicated boilerplate code now to enable the tick icon, which would be much nicer pulled out into a separate component, but I don't mind merging this anyway.

**Review by aptkingston — COMMENTED** [review]
Nice @ConorWebb96 - much cleaner!

Not a blocker, but the cleaner way to structure these files would be create a folder `DataSourceSelect`, then put both the existing `DataSourceSelect.svelte` inside that, and also this new component `SelectItem.svelte`. I'd also call the component something like `DataSourceCategory` 👌

File: packages/builder/src/components/design/settings/controls/DataSourceSelectItem/SelectItem.svelte [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  {#each dataSet as data}
+    <li
+      class="spectrum-Menu-item"
+      class:is-selected={value?.label === data.label &&
+        value?.type === data.type}
+      role="option"
+      aria-selected="true"
+      tabindex="0"
+      on:click={() => onSelect(data)}
+      on:keydown={e => {
```
[the remark this answers is not in the document]
Comment (ConorWebb96) [reply packages/builder/src/components/design/settings/controls/DataSourceSelectItem/SelectItem.svelte]: Fixed thanks for the feedback.

**Comment by aptkingston** [comment]
Looks nearly ready to go @ConorWebb96 👍 Tested this in the feature branch and just noticed a small style issue - the options don't use the full width of the popover (notice the grey at the right side edge):
image

Completely not a blocker but could we also name the folder just `DataSourceSelect` instead of `DataSourceSelectItem`? That's the pattern we normally use.

**Pull request merged**